modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolderHeader/ModalPublishObjectDefinitions.tsx

```diff
@@ -0,0 +1,519 @@
+	);
+	const [
+		publishObjectDefinitionsStatus,
+		setPublishObjectDefinitionsStatus,
+	] = useState<number>(STATUS.DRAFT);
+	const [
+		selectAllDraftObjectDefinitions,
+		setSelectAllDraftObjectDefinitions,
+	] = useState<boolean>(false);
+	const [
+		selectedDraftObjectDefinitions,
+		setSelectedDraftObjectDefinitions,
+	] = useState<SelectedDraftObjectDefinition[]>([]);
+
+	const updateObjectDefinitionStatus = (
+		selectedDraftObjectDefinitions: SelectedDraftObjectDefinition[],
+		objectDefinitionId: number,
+		objectDefinitionStatus: ObjectDefinitionStatus,
+		errorMessage?: string
+	) => {
+		return selectedDraftObjectDefinitions.map(
+			(selectedDraftObjectDefinition) => {
+				if (selectedDraftObjectDefinition.id === objectDefinitionId) {
+					return {
+						id: objectDefinitionId,
+						status: objectDefinitionStatus,
+						...(objectDefinitionStatus.code === STATUS.REJECTED && {
+							errorMessage,
+						}),
+					};
+				}
+				else {
+					return selectedDraftObjectDefinition;
+				}
+			}
+		) as SelectedDraftObjectDefinition[];
+	};
+
+	const publishObjectDefinition = (
+		objectDefinitionId: number
+	): Promise<ObjectDefinition | number> => {
+		// eslint-disable-next-line no-async-promise-executor
+		return new Promise<ObjectDefinition | number>(async (resolve) => {
+			try {
+				const objectDefinitionResponse = await API.postObjectDefinitionPublish(
+					objectDefinitionId
+				);
+
+				const objectDefinitionResponseJSON = await objectDefinitionResponse.json();
+
+				if (!objectDefinitionResponse.ok) {
+					throw new Error(objectDefinitionResponseJSON.title);
+				}
+
+				setSelectedDraftObjectDefinitions((prevState) =>
+					updateObjectDefinitionStatus(
+						prevState,
+						objectDefinitionId,
+						objectDefinitionResponseJSON.status
+					)
+				);
+
+				resolve(objectDefinitionResponseJSON);
+			}
+			catch (error: any) {
+				setSelectedDraftObjectDefinitions((prevState) =>
+					updateObjectDefinitionStatus(
+						prevState,
+						objectDefinitionId,
+						{code: STATUS.REJECTED},
+						error.message
+					)
+				);
+
+				// don't throw reject, so that it doesn't go to the catch flow of the promise.all
+
+				resolve(STATUS.REJECTED);
+			}
+		});
+	};
+
+	const handleOnClickPublish = async () => {
+		setModalHeaderMessage(`${Liferay.Language.get('publishing')}...`);
+		setPublishObjectDefinitionsStatus(STATUS.PENDING);
+
+		const publishPromises = selectedDraftObjectDefinitions.map(
+			({id, status}) => {
+				setSelectedDraftObjectDefinitions((prevState) =>
+					updateObjectDefinitionStatus(prevState, id, status)
+				);
+
+				return publishObjectDefinition(id);
+			}
+		);
+
+		try {
+			const responses = await Promise.all(publishPromises);
+
+			const hasRejectedResponses = responses.some(
+				(response) =>
+					typeof response === 'number' && response === STATUS.REJECTED
+			);
+			const filteredResponses = responses.filter(
+				(response) => typeof response === 'object'
+			);
```

Comment:
hey @naraandrade, when I read this code snippet, it wasn't clear to me what value the filteredResponses constant stores.

It is constructed using response, which in turn is constructed using publishPromises.

I need to think and analyze the code to understand what this code stores. Do you think we can improve the name of this variable to make this code easier to understand?